src/python/pants/backend/terraform/target_types.py

```diff
@@ -51,5 +57,71 @@ class TerraformModuleTarget(Target):
     )
 
 
+class TerraformBackendConfigField(OptionalSingleSourceField):
+    alias = "backend_config"
+    help = "Configuration to be merged with what is in the configuration file's 'backend' block"
+
+    def empty(self) -> TerraformBackendConfigField:
+        """Clear the backend config to ensure that."""
+        return TerraformBackendConfigField(None, self.address)
+
+
+class TerraformVarFilesField(MultipleSourcesField):
```

Comment:
Maybe `TerraformVarFileSourcesField`? It's just a nit, but in my head the `Sources` bit helps understand this isn't any kind of value holder field, but a descendant of a `SourceField`.

Reply:
makes sense, renamed